libs/platform/src/lib/approval-flow/approval-flow-node/approval-flow-node.component.ts

```diff
@@ -80,6 +80,14 @@ export class ApprovalFlowNodeComponent implements OnInit, OnChanges, OnDestroy {
      */
     @Input() allNodesInColumnApproved = false;
 
+    /** A reference to the app custom statuses */
+    @Input()
+    approvalStatusTemplate: TemplateRef<any>;
```

Comment:
This works, but why not just accept string inputs? Would be much simpler

Reply:
@Mike O'Donnell, the 9630
>This works, but why not just accept string inputs? Would be much simpler
>Do you mean to pass an array? The reason for the TemplateRef is 

The idea behind this was

1.  you have this template there and you need to provide also internationalisation and  using a pipe it might be easier than just to maintain different versions of strings in your dataset, and I wanted to keep it in sync to have the same behaviour what we have inside

```html
  <ng-container [ngSwitch]="node.status">
            <ng-container *ngSwitchCase="'approved'">{{
                'platformApprovalFlow.statusApproved' | fdTranslate
            }}</ng-container>
            <ng-container *ngSwitchCase="'rejected'">{{
                'platformApprovalFlow.statusRejected' | fdTranslate
            }}</ng-container>
            <ng-container *ngSwitchCase="'in progress'">{{
                'platformApprovalFlow.statusInProgress' | fdTranslate
            }}</ng-container>
            <ng-container *ngSwitchCase="'not started'">{{
                'platformApprovalFlow.statusNotStarted' | fdTranslate
            }}</ng-container>
        </ng-container>

```
2. You want to map your STATUS -> ObjectStatus colors. 

3. You have a `ApprovalNode` instance, which is passed from `ApprovalFlowComponent` (when rendering every node), you want to pass this object to the app to decide how they want to render this status.

4. Extensibility. If they want to provide really custom behavior for the status, like they want to concatenate something or use icons maybe they can here. 


This way you are leaving it completly on the application and you dont need to add additional functionality into the `ApprovalFlowComponent`.

I can imagine you can pass something like 

```ts
  @Input
  approvalStatusMapping: Record< ApprovalStatus, strings>
```
And internally you would have to make sure when rendering a node, you check the strings + add additional piece of code that handles this mappings. The problem I see if you pass already translated string literals into it like `日期范围输入 | approved`, treating i18n data already as part of the app data instead of static strings I am not sure if this would not lead to problems. So I believe this is more saver and flexible. 

Let me know your thought...

Reply:
This is much more flexible and robust, and I agree with Frank